src/OpenSSL/_util.py

```diff
@@ -96,12 +97,17 @@ def path_string(s):
 
     :return: An instance of :py:class:`bytes`.
     """
-    if isinstance(s, bytes):
-        return s
-    elif isinstance(s, str):
-        return s.encode(sys.getfilesystemencoding())
+    try:
+        strpath = os.fspath(s)
+    except Exception:
```

Comment:
Why do we need to catch exceptions here? What case is that handling?

Reply:
```
In [2]: os.fspath(1)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-b00c080840d9> in <module>
----> 1 os.fspath(1)

TypeError: expected str, bytes or os.PathLike object, not int
```

Reply:
Just because you seem to want to provide your own message. I'd be fine to not handle it and remove the else case later on as well.

Reply:
Yes, I think allowing the exception from `os.fspath` to propagate is fine.